crates/swc_ecma_transforms_typescript/src/strip.rs

```diff
@@ -985,15 +1012,33 @@ where
             .collect::<Result<Vec<_>, _>>()
             .unwrap_or_else(|_| panic!("invalid value for enum is detected"));
 
-        let is_all_str = members
-            .iter()
-            .all(|(_, v)| matches!(v, Expr::Lit(Lit::Str(..))));
+        let contains_lit = members.iter().any(|(_, v)| matches!(v, Expr::Lit(..)));
+        let should_collect = self.config.ts_enum_config.should_collect_enum(e.is_const);
+
+        if should_collect && contains_lit {
+            let mut enums = self.ts_enum_lit.borrow_mut();
+
+            let k_v = enums.entry(e.id.to_id()).or_default();
+
+            members
+                .iter()
+                .filter(|(_, v)| matches!(v, Expr::Lit(..)))
+                .for_each(|(m, e)| {
+                    let member_key = m.id.as_ref().clone();
+                    let value = e.clone().expect_lit();
```

Comment:
Can you add a test for enum which has a template literal in its initializer?

Reply:
Done.
see:
https://github.com/swc-project/swc/blob/ac1f5f97f0d801608994b84ff7b51a194d3a2174/crates/swc/tests/fixture/ts-inline-enum/default/input/index.ts#L11-L18
https://github.com/swc-project/swc/blob/ac1f5f97f0d801608994b84ff7b51a194d3a2174/crates/swc/tests/fixture/ts-inline-enum/default/output/index.ts#L10-L17